lib/improver/utilities/cube_metadata.py

```diff
@@ -207,6 +207,11 @@ def update_coord(cube, coord_name, changes, warnings_on=False):
                    "{}".format(coord_name))
             warnings.warn(msg)
     else:
+        # Unit conversion must take place prior to potentially setting
+        # points and bounds, so that unit conversion does not conflict with
+        # setting points and bounds.
```

Comment:
This raises a good point about changing units, points and bounds that worries me for a couple of reasons: 

1. This is a change to behaviour that isn't covered by any of the tests
1. This behaviour is arguably unsafe - how does the user define the correct behaviour when creating their dictionary?
1. This behaviour is probably unnecessary (I think the only situation in which we would want to set all of units, points and bounds is when we are adding a new coordinate, not updating an existing one).

I'd suggest that `update_coord` should fail if `units` AND either of `points` or `bounds` are set, as we should not need these functions simultaneously.

Reply:
I've updated the behaviour of `update_coord` to raise an exception as you've suggested.